src/Plugins/Reminder.php

```diff
@@ -0,0 +1,357 @@
+          )
+          \b
+      )
+    )
+    
+    ^(?<time>(?:(?&timepart)(*SKIP).)+)(?<string>.+)$
+    /uix
+REGEX;
+
+
+    public function __construct(ChatClient $chatClient, KeyValueStore $storage, AdminStore $admin, array $watchers = []) {
+        $this->chatClient = $chatClient;
+        $this->watchers = $watchers;
+        $this->storage = $storage;
+        $this->admin = $admin;
+    }
+
+    private function flush(Command $command) {
+        return resolve(function() use($command) {
+            if (!yield $this->admin->isAdmin($command->getRoom(), $command->getUserId())) {
+                return $this->chatClient->postReply($command, "I'm sorry Dave, I'm afraid I can't do that");
+            }
+
+            $reminders = yield $this->storage->getKeys($command->getRoom());
+            if ($reminders !== []) {
+                foreach ($reminders as $key){
+                    yield $this->storage->unset($key, $command->getRoom());
+                }
+            }
+            return $this->chatClient->postMessage($command->getRoom(), "Reminders are gone.");
+        });
+    }
+
+    private function unset(Command $command) {
+        $messageId = $command->getParameter(1) ?? false;
+
+        return resolve(function() use($command, $messageId) {
+            if (!yield $this->admin->isAdmin($command->getRoom(), $command->getUserId())) {
+                return $this->chatClient->postReply($command, "I'm sorry Dave, I'm afraid I can't do that");
+            }
+
+            if(!$messageId){ return $this->chatClient->postMessage($command->getRoom(), self::USAGE); }
+            $key = $messageId;
+
+            if (yield $this->storage->exists($key, $command->getRoom())) {
+                yield $this->storage->unset($key, $command->getRoom());
+                return $this->chatClient->postMessage($command->getRoom(), "Reminder unset.");
+            }
+            return $this->chatClient->postReply($command, "I'm sorry, I couldn't find that key.");
+        });
+    }
+
+    private function setReminder(Command $command): Promise
+    {
+        $commandName = $command->getCommandName();
+        return resolve(function() use($command, $commandName) {
+
+            switch ($commandName){
+                case 'in':
+                    if(preg_match(self::TIME_STRING_REGEX, implode(" ", $command->getParameters()), $matches)){
+                        $time = $matches['time'] ?? false;
+                        $text = $matches['string'] ?? false;
+                    }
+                    break;
+                case 'at':
+                    $time = $command->getParameter(0) ?? false; // 24hrs
+                    if($time && preg_match(self::TIME_FORMAT_REGEX, $time)){ // !!at 16:00 remind | !!at monday next week remind?
+                        $text = implode(" ", array_diff($command->getParameters(), array($time)));
+                    }
+                    break;
+                case 'reminder':case 'remind':
```

Comment:
Remove hardcoded `remind` case and add it as an alias using the `!!command alias` thing instead